File: K_radius_subarray_averages.cpp
//Ques: 2090 K Radius Subarray Averages

// class Solution {
// public:
//     vector<int> getAverages(vector<int>& nums, int k) {
//         int n = nums.size();
//         vector<int> ans(n,-1);

//         long long s=0;
//         for (int i=0; i<n; i++) {
//             s = s + nums[i];
//         }
//         if(i >= k+2) {
//             ans[i-k] = s/ 2*k+1;

//             s = s- nums[i-2k];
//         }

//     }
// };




class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {

        int n = nums.size();
        vector<int> ans(n, -1);

        long long s = 0;

        for (int i = 0; i < n; i++) {

            s = s + nums[i];

            if (i >= 2 * k + 1) {
                s = s - nums[i - (2 * k + 1)];
            }

            if (i >= 2 * k) {
                ans[i - k] = s / (2 * k + 1);
            }
        }

        return ans;
    }
};

File: K Radius Subarray Averages.py

#2090. K Radius Subarray Averages

class Solution:
    def getAverages(self, nums: list[int], k: int):

        n = len(nums)

        ans = [-1] * n

        window = 2 * k + 1

        if window > n:
            return ans

        curr_sum = sum(nums[:window])

        ans[k] = curr_sum // window

        for i in range(window, n):

            curr_sum += nums[i]
            curr_sum -= nums[i - window]

            center = i - k

            ans[center] = curr_sum // window

        return ans  